src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -203,10 +203,18 @@ public IEnumerable<TagHelperRequiredAttributeDescriptor> RequiredAttributes
         /// </summary>
         public TagHelperDesignTimeDescriptor DesignTimeDescriptor { get; set; }
 
-        public Dictionary<string, string> PropertyBag
+        /// <summary>
+        /// A dictionary containing additional properties of the tag helper.
+        /// </summary>
+        public IDictionary<string, string> PropertyBag
         {
             get
             {
+                if (_propertyBag == null)
+                {
+                    _propertyBag = new Dictionary<string, string>();
```

Comment:
You might want to specify the casing behavior for the dictionary. In general most of our things use `OrdinalIgnoreCase` for dictionaries.